lib/charliecloud.py

```diff
@@ -739,8 +789,17 @@ def get_layer(self, hash_, path):
    def get_manifest(self, path):
       "GET the manifest for the image and save it at path."
       url = self._url_of("manifests", self.ref.version)
-      accept = "application/vnd.docker.distribution.manifest.v2+json"
-      self.get(url, path, { "Accept": accept })
+      sv = {'1': "application/vnd.docker.distribution.manifest.v1+json",
+            '2': "application/vnd.docker.distribution.manifest.v2+json"}
+      # Use undocumented variable to specify schema version for testing.
+      if env_schema_version:
+            DEBUG('requesting schema version %s (environment)'
+                   % env_schema_version)
+            accept = sv[env_schema_version]
+      else:
+         accept = [sv['2'], sv['1']]
+      self.get(url, path, { "Accept": str(accept) })
+
```

Comment:
1. Can we add weights to prefer v2 over v1 if we accept both?
2. If you write the logic like the following, I think it will be simpler:
```
sv = {}
if v1 is OK:
  sv['1'] = ...
if v2 is OK:
  sv['2'] = ...
assert (sv is not empty)
```